src/typescript/azure-openapi-validator/rules/EnumMustHaveType.ts

```diff
@@ -0,0 +1,22 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from "../rule";
+export const EnumMustHaveType: string = "EnumMustHaveType";
+
+rules.push({
+  id: "R3015",
+  name: EnumMustHaveType,
+  severity: "warning",
+  category: "ARMViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm | OpenApiTypes.dataplane,
+  appliesTo_JsonQuery: "$..*[?(@.enum)]",
+  run: function*(doc, node, path) {
+    const msg: string = `Enum must define its type and "object" type is not allowed due to Autorest refuse to parse it.`;
+    if (node.type === undefined || node.type.toLowerCase() === "object") {
```

Comment:
could you use !node.type instead?

Reply:
I perfer ”node.type===undefined“ is more clear and means author forget to define its type. `!node.type` include `false, empty, null...`.

Reply:
what is the type of node.type, is it string?
what if node.type is null?

Reply:
type is string.